### ISSUE TITLE
Review costing for unnecessary use of vectors

### ISSUE THREAD

**Issue by ianthetechie** [valhalla/valhalla#4505]
This came up while reviewing #4492, where someone flagged the oddity of using a vector for the speed factor, which could be represented more efficiently as a fixed size array. This is an issue is common across all costing profiles.

**Comment by nilsnolde** [comment]
Yep, even more actually, it should be a compile time constant in DynamicCost since it's all static data and the same for each costing model.
